packages/atlas-service/src/util.ts

```diff
@@ -11,16 +11,113 @@ export type IntrospectInfo = { active: boolean };
 
 export type Token = plugin.IdPServerResponse;
 
+function hasExtraneousKeys(obj: any, expectedKeys: string[]) {
+  return (
+    Object.keys(obj).filter((key) => !expectedKeys.includes(key)).length > 0
```

Comment:
tiny and very optional suggestion:

```suggestion
    Object.keys(obj).some((key) => !expectedKeys.includes(key))
```